go-controller/pkg/ovn/gateway_init.go

```diff
@@ -560,16 +560,16 @@ func (oc *DefaultNetworkController) addExternalSwitch(prefix, interfaceID, nodeN
 	return nil
 }
 
-func (oc *DefaultNetworkController) addPolicyBasedRoutes(nodeName, mgmtPortIP string, hostIfAddr *net.IPNet, otherHostAddrs []string) error {
+func (oc *DefaultNetworkController) addPolicyBasedRoutes(nodeName, mgmtPortIP string, hostIfCIDR *net.IPNet, otherHostAddrs []string) error {
```

Comment:
NOTE: This path of code is not tested at all which is why this bug was not caught when the match of the policy changed from hostIP to hostIP/16. No code should be able to modify core routing policies on ovn_cluster_router and get away with it. Tests should complain.
```
matchstr2 := fmt.Sprintf(`inport == "rtos-%s" && ip4.dst == nodePhysicalIP /* %s */`, nodeName, nodeName)
matchstr3 := fmt.Sprintf("ip4.src == source && ip4.dst == nodePhysicalIP")
matchstr6 := fmt.Sprintf("ip4.src == NO DELETE && ip4.dst == nodePhysicalIP /* %s-no */", nodeName)
```
All current unit tests are playing with dummy match expresssions trying to leverage the comment on the actual policy:

We need unit test coverage in OVN DB libovsdb test framework in gateway_test.go and gateway_init.go for the match to prevent this from happening in the future. I will not block this PR for that but I will open an issue for this.

Reply:
hmm not true, we do have some tests here: `addNodeLogicalFlows` however for nonIC setups these should have clearly failed - we got lucky with upgrades for ic only because db's are wiped, so nonIC->nonIC should have been stuck forever but it didn't unsure why.